haystack/components/agents/agent.py

```diff
@@ -558,31 +558,19 @@ async def run_async(  # noqa: PLR0915
             )
             raise ValueError(msg)
 
-        self._agent_name = self.__component_name__ if hasattr(self, "__component_name__") else "isolated_agent"
-
         # validate breakpoints
         if break_point and isinstance(break_point.break_point, ToolBreakpoint):
             self._validate_tool_breakpoint_is_valid(break_point)
 
         # Handle resume state if provided
         if resume_state:
-            # Extract component visits from pipeline state
             component_visits = resume_state.get("pipeline_state", {}).get("component_visits", {})
-
-            # Initialize with default values if not present in resume state
-            component_visits = dict.fromkeys(["chat_generator", "tool_invoker"], 0) | component_visits
-
-            # Extract state data from pipeline state
```

Comment:
This was removed to match the same implementation in `Agent.run`. Let me know if this should be added back and also added to the `run` method.

Reply:
also for coherence with the sync method